jetty-http3/http3-http-client-transport/src/main/java/org/eclipse/jetty/http3/client/http/internal/HttpReceiverOverHTTP3.java

```diff
@@ -59,9 +59,24 @@ protected void receive()
             return;
 
         if (notifySuccess)
+        {
             responseSuccess(exchange);
+        }
         else
-            getHttpChannel().getStream().demand();
+        {
+            Stream stream = getHttpChannel().getStream();
+            if (LOG.isDebugEnabled())
+                LOG.debug("Demanding from {} in {}", stream, this);
+            if (stream == null)
+                return;
```

Comment:
Ok for this additional check, but did you remove the similar check in `read(boolean)` that was in #9756?

Reply:
Done.

Reply:
No, I did not. This change won't merge to 12 and my plan is to replace this change with the `if (stream == null) return Content.Chunk.from(new EOFException())` check.